src/storage-controller/src/lib.rs

```diff
@@ -1503,7 +1512,16 @@ where
         StorageError<Self::Timestamp>,
     > {
         if self.read_only {
-            return Err(StorageError::ReadOnly);
+            if !commands.iter().all(|(id, _)| id.is_system()) {
+                // TODO: we need to drop writes for non-migrated system tables,
+                // or very bad things will happen, as we'll be writing directly
+                // to shards that are still managed by txn-wal in the old
+                // generation.
```

Comment:
XXX: flagging this is a TODO to address before we merge this PR. (It may already have been on your radar, Joe, but I wanted to be sure to call it out just in case.)